content/news/2023-03-04-bevy-0.10/index.md

```diff
@@ -0,0 +1,1682 @@
+
+With the addition of the new `Schedules` resource and the `world.run_schedule(schedule_label: impl ScheduleLabel)`API it's more ✨ ergonomic ✨ than ever.
+
+```rust
+// A schedule!
+let mut my_schedule = Schedule::new();
+schedule.add_system(my_system);
+
+// A schedule label for it
+#[derive(ScheduleLabel, Debug, Hash, PartialEq, Eq, Clone)]
+struct MySchedule;
+
+// An exclusive system to run this schedule
+fn run_my_schedule(world: &mut World) {
+    while very_complex_logic(){
+        world.run_schedule(MySchedule);
+    }
+}
+
+// Behold the ergonomics
+app
+    .add_schedule(MySchedule, my_schedule)
+    .add_system(run_my_schedule);
+```
+
+Bevy uses this pattern for five rather different things at 0.10 release:
+
+1. **Startup systems:** these now live in their own schedule, which is run once at the start of the app.
+2. **Fixed timestep systems:** another schedule?! The exclusive system that runs this schedule accumulates time, running a while loop that repeatedly runs `CoreSchedule::FixedTimestep` until all of the accumulated time has been spent.
+3. **Entering and exiting states:** a bonanza of schedules. Each collection of systems that runs logic to enter and exit a state variant is stored in its own schedule, which are called based on the change in state in the `apply_state_transitions::<S>` exclusive system.
+4. **Rendering:** all rendering logic is stored in its own schedule to allow it to run asynchronously relative to gameplay logic.
+5. **Controlling the outermost loop:** in order to handle the "startup schedule first, then main schedule" logic, we wrap it all up in a minimal overhead `CoreSchedule::Outer` and then run our schedules as the sole exclusive system there.
+
+Follow the breadcrumbs starting at [`CoreSchedule`](https://dev-docs.bevyengine.org/bevy/app/enum.CoreSchedule.html) for more info.
+
+### Simpler Run Conditions
+
+Systems may have any number of run conditions (and inherit them from the sets they belong to), but will only run if all of their run conditions return `true`.
+Run criteria have been renamed to the clearer **run conditions**, which can be constructed out of any read-only system that returns `bool`.
+
+With a new blessed pattern for complex control flow, we can finally get rid of looping run criteria. [`ShouldRun::YesAndCheckAgain`](https://docs.rs/bevy/0.9.1/bevy/ecs/schedule/enum.ShouldRun.html) was not exactly straightforward to reason about, either for engine devs or users. It's always a bad sign when your bool-like enums have four possible values.
+
+If you crave that powerful, complex control flow: use the "schedules in exclusive systems" pattern listed above.
+For the other 99% of use cases, enjoy the simpler `bool`-based run conditions.
+
+```rust
+// Let's make our own run condition
+fn game_end_condition(query: Query<&Player>, score: Res<Score>) -> bool {
+    let player = query.single();
+    player.is_alive() && score.0 > 9000
+}
+
+app.add_system(win_game.run_if(game_end_condition));
```

Comment:
Upon reading `fn game_end_condition` I assumed it was a loss condition and was confused until seeing the `win_game` system right at the end.

Renaming to `game_win_condition` makes the intent clear from the beginning, but is a bit repetitive.

Reply:
```suggestion
fn game_win_condition(query: Query<&Player>, score: Res<Score>) -> bool {
    let player = query.single();
    player.is_alive() && score.0 > 9000
}

app.add_system(win_game.run_if(game_win_condition));
```